File: ios/BasicQRScanner/Vision/VisionViewManager.m
//
//  VisionViewManager.m
//  BasicQRApp
//
//  Created by Ananth Desai on 02/01/25.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(VisionViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(onSuccessfulScan, RCTDirectEventBlock)
@end

File: ios/BasicQRScanner/Vision/VisionViewManager.swift
//
//  VisionViewManager.swift
//  BasicQRApp
//
//  Created by Ananth Desai on 02/01/25.
//

import Foundation

@objc(VisionViewManager)
class VisionViewManager: RCTViewManager {
  private var scannerRef: VisionView? = nil
  
  override class func requiresMainQueueSetup() -> Bool {
    false
  }
  
  override func view() -> UIView! {
    let ref = VisionView()
    scannerRef = ref
    return ref
  }
}
